### PR TITLE
fix: sync @typescript-eslint versions across repo

### LINKED ISSUE #555: bug: latest version incorrectly detects imported types as unused vars
Hey folks! Running into an issue with the latest version of web-scripts, likely due to one or more of the packages that were indirectly upgraded. After this update, the linter now complains about imported types as unused vars. Here's a repository: https://github.com/mtlewis/web-scripts-repro that demonstrates the issue - try cloning and running `yarn lint`.

### PR DESCRIPTION
**Pull request by mtlewis**
Closes #555.

Having different versions of `@typescript-eslint/parser` and `@typescript-eslint/eslint-plugin` seems to cause false positives like the one described in #555. This commit syncs the versions which appears to resolve the issue.

### CODE REVIEW

**Comment by Rugvip** [comment]
Heads up that we tried bumping @typescript-eslint in https://github.com/spotify/backstage but had a bunch of breakages. They were a bit all over the place, both complaints about unused `React` imports and usages of undefined `NodeJS` global types. We're holding off the bumping for now, but that's mainly just to keep focus elsewhere. The issues seemed like things fixable through updating eslint configuration and avoiding duplicate `@types/*` packages.

In short: you may want to do a major version bump and provide some guidance for users.

**Comment by mtlewis** [comment]
Thanks @Rugvip - yep, definitely seems to make sense to consider both upgrading and downgrading to sync these packages.

**Comment by mtlewis** [comment]
@javoire any thoughts on this? I'm happy to prep a PR the other way if you prefer downgrading to sync these packages, but it does seem like things are in a broken state now so we do probably need one fix or the other here.

**Comment by javoire** [comment]
How does it relate to this PR https://github.com/spotify/web-scripts/pull/519/files ? I see a few similarities, and that's planning for a major release. Should we combine

**Comment by mtlewis** [comment]
Seems totally reasonable to include this in the next major! I would point out though that given the minimal repro in #555, it seems likely that it is affecting most people that are using web-scripts with typescript, so if #519 is still a ways off, perhaps it could make sense to try and get this out sooner.

**Comment by javoire** [comment]
@mtlewis we decided to bake it into v9, cutting a release very soon!
